### PR TITLE
Bug fix, tests, and assertion for stall condition

### PR DESCRIPTION
**Pull request by AaronGChan**
Fixing https://github.com/riscv-software-src/riscv-perf-model/issues/116

### CODE REVIEW

File: core/Rename.cpp [hunk 1]
```diff
@@ -356,4 +359,9 @@ namespace olympia
         }
     }
 
+
+    void Rename::onStartingTeardown_(){
+        sparta_assert(current_stall_ == StallReason::NOT_STALLED, "Rename is stalled when entering teardown, possible lockup in Rename?")
```
Comment (AaronGChan) [inline core/Rename.cpp:364]: @knute-sifive Was thinking either to have an assert like this to ensure that there is no ending stall when teardown begins, or should I do a notification similar to https://github.com/riscv-software-src/riscv-perf-model/pull/108

Comment (knute-sifive) [reply core/Rename.cpp:364]: Yeah, you can't have this assertion.  It's possible that Rename is stalled waiting for free registers AND the ROB was told to stop simulation on instruction _N_.  This is a valid case and you don't want to assert on that. 

In #108 (still under review at this time), having the ROB broadcast "I stopped simulation" and Rename listening for that and determining if it's a cause in this function is appropriate.

File: core/Rename.cpp [hunk 2]
```diff
@@ -132,18 +132,21 @@ namespace olympia
         }
 
         const auto & srcs = inst_ptr->getRenameData().getSourceList();
-        // decrement reference for store instruction to data register
-        if(inst_ptr->isStoreInst()) {
+        // decrement reference to data register
+        if(inst_ptr->isLoadStoreInst()){
```
Comment (knute-sifive) [inline core/Rename.cpp:136]: @danbone's fix is a good catch: noticing that the renaming and reclamation are not in sync

But there's something nagging me about all of this -- we're renaming `x0`.  Can you open a new issue that fixes that?

Specifically, in `renameInstructions_`, when iterating over the sources and destinations, you need to _skip_ renaming `x0`:  

```
if (num == 0 && rf == core_types::RF_INTEGER) { continue; }
```

**Review by knute-sifive — APPROVED** [review]
I'll approve this for now, but please put in a proper fix for the end-of-simulation assertion (use the ROB notification).

File: core/Rename.cpp (hunk 1 above)
Comment (AaronGChan) [reply core/Rename.cpp:364]: I'll remove the teardown and assertion for now, and I'll add in the rename debugging once https://github.com/riscv-software-src/riscv-perf-model/pull/108 is merged.

**Pull request merged**